src/mbgl/renderer/render_orchestrator.cpp

```diff
@@ -337,6 +337,11 @@ std::unique_ptr<RenderTree> RenderOrchestrator::createRenderTree(
                 renderItemsEmplaceHint = layerRenderItems.emplace_hint(renderItemsEmplaceHint, layer, nullptr, index);
             }
         }
+
+        if (auto sourcePrefetchDelta = sourceImpl->getPrefetchZoomDelta()) {
```

Comment:
Could this code be inside Render::SourceUpdate?

Reply:
Update is implemented by concrete classes, I pushed implementation down to rendersource::update and added parameter to pyramid update. https://github.com/mapbox/mapbox-gl-native/pull/16179/commits/c7ab18af52087f1e483f91c4037248429460643a